src/main/java/searching/Tree23.java

```diff
@@ -0,0 +1,261 @@
+package searching;
+
+/**
+ * In this exercise, we ask you to implement the put method of a 23Tree.
+ * To do so, we recommend you to divide the task in 2 different functions : put2Node and put3Node.
+ * The first one can be used to put a value in a 2Node and the second one in a 3Node.
+ *
+ * We strongly recommend you to divide this task to make the implementation easier.
+ * We also give you 3 static variables (accessible everywhere in this file), which
+ * you can use to store intermediate result of the put method
+ * (for example if you must insert a key higher in tree after manipulations).
+ * The two other variables can be used to store temporary subtrees.
+ *
+ * Also don't forget that if a key is taken up to the root and the root is already
+ * a 3Node, you must create a new level in the tree
+ */
+public class Tree23 {
+
+    Node23 root;
+
+    public Tree23(){
+        root = null;
+    }
+
+    static Node23 carry;
+    static Node23 subtree, subtree2;
+
+    public Node23 put(Node23 n, Integer key, Double value){
+        if (n == null) return new Node23(key, value, null, null);
+        Node23 ret = n.put(key, value);
+        if (carry != null) {
+            carry.left = subtree;
+            carry.middle = subtree2;
+            return carry;
+        }
+        return ret;
+    }
+
+    public void put(Integer key, Double value){
+        // BEGIN STRIP
+        carry = null;
+        root = put(root, key, value);
+        // END STRIP
+        // TODO
+        // STUDENT return;
+
+    }
+
+    public Double search(Integer k){
+        if (root == null) return null;
+        Double temp = root.search(k);
+        return temp;
+    }
+
+    public boolean isBalanced(){
+        if (root == null) return true;
+        return root.isBalanced() >= 0;
+    }
+
+    public boolean isBst(){
+        if (root == null) return true;
+        return root.isBst(Integer.MIN_VALUE, Integer.MAX_VALUE);
+    }
+    
+}
+
+class Node23{
+
+    Integer k1, k2;  // k1 < k2
+    Double v1, v2;
+    Node23 left, middle, right;
```

Comment:
Named `leftChild`, `centerChild` and `rightChild` to explicitely show that these are the children of the node